invoices/views.py

```diff
@@ -96,12 +99,90 @@ def invoice_new_confirmation(request, pk):
         invoice = Invoice.objects.get(pk=int(pk))
 
     except ObjectDoesNotExist:
-        messages.error(request, 'Ett fel uppstod och kvittot skapades inte.')
+        messages.error(request, 'Ett fel uppstod och fakturan skapades inte.')
         return HttpResponseRedirect(reverse('invoices-new'))
 
     return render(request, 'invoices/confirmation.html', {'invoice': invoice})
 
+@login_required
+@require_http_methods(["GET", "POST"])
+def edit_invoice(request, pk):
+    """
+    Shows form for editing invoices.
+    """
+    try:
+        invoice = Invoice.objects.get(pk=int(pk))
+    except ObjectDoesNotExist:
+        raise Http404("Fakturan finns inte")
+    
+    if not request.user.profile.may_delete_invoice(invoice):
+        return HttpResponseForbidden('Du har inte behörighet att redigera denna faktura.')
+    
+    if request.method == 'GET':
+        return render(request, 'invoices/edit.html', {
+            "invoice": invoice,
+            "invoice_parts": invoice.invoicepart_set.all(),
+            "budget_url": settings.BUDGET_URL,
+            })
+    
+    invoice.description = request.POST['description']
+    invoice.invoice_date = request.POST['invoice-date']
+    invoice.due_date = request.POST['due-date']
+    invdate = invoice.invoice_date
+    duedate = invoice.due_date
+    invoice.save()  #Saves the change to invoice_date and due_date
+    
+    # Beneficial to not return per check, because then several errors can be reported at once
+    valid = True
+    """
```

Comment:
Do you mean:
```
if len((request.FILES.getlist('files'))) < 1:
        messages.error(request, 'Du måste ladda upp en fil med fakturan')
        valid = False
```
It is used by the other file upload method, but doesn't seem to work for this one.

Reply:
Fixed by changing to:
```
if len((request.FILES.getlist('files'))) < 1 and len((request.POST.getlist('fileIds[]'))) < 1:
        messages.error(request, 'Du måste ladda upp en fil med fakturan')
        valid = False
```

Reply:
Might not be needed since the submit button checks it as well

Reply:
yes but it doesn't hurt to check on the backend as well (even though it's not a security issue, since you'll just not get it attested/confirmed/paid if there are no files)